src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -308,39 +317,40 @@ private static void CheckParenthesizedLambdaParameters(
         {
             if (parameterSyntaxList.Count > 0)
             {
-                var hasTypes = parameterSyntaxList[0].Type != null;
+                // If one parameter has a type, then all parameters must have a type.
+                var requiresTypes = parameterSyntaxList.Any(static p => p.Type != null);
 
-                checkForImplicitDefault(hasTypes, parameterSyntaxList[0], diagnostics);
-
-                for (int i = 1, n = parameterSyntaxList.Count; i < n; i++)
+                foreach (var parameter in parameterSyntaxList)
                 {
-                    var parameter = parameterSyntaxList[i];
-
                     // Ignore parameters with missing names.  We'll have already reported an error
                     // about them in the parser.
                     if (!parameter.Identifier.IsMissing)
                     {
-                        var thisParameterHasType = parameter.Type != null;
-
-                        if (hasTypes != thisParameterHasType)
+                        if (requiresTypes)
                         {
-                            diagnostics.Add(ErrorCode.ERR_InconsistentLambdaParameterUsage,
-                                parameter.Type?.GetLocation() ?? parameter.Identifier.GetLocation());
+                            if (parameter.Type is null)
+                            {
+                                diagnostics.Add(ErrorCode.ERR_InconsistentLambdaParameterUsage,
+                                    parameter.Identifier.GetLocation());
+                            }
+                        }
+                        else
```

Comment:
i dont feel like that that is that valuable.  The user will already get a good message saying that they need to provide a type since some other parameter has a type.

So the only way to get in here is if none of hte parameters has a type, at which point it is reasonable to tell them they can't have a default.

Reply:
put another way: when there are multiple problems going on, it doesn't bother me that you don't hear about all of them at once. especially as it may just be an initial accident (like deleting the type name from one unintentionally).  Getting several errors at once, instead of an initial clear error, is not preferable for me.